hikari/templates.py

```diff
@@ -174,5 +175,174 @@ class Template:
     is_unsynced: bool = attr.field(eq=False, hash=False, repr=False)
     """Whether this template is missing changes from it's source guild."""
 
+    async def fetch_self(self) -> Template:
+        """Fetch an up-to-date view of this template from the API.
+
+        Returns
+        -------
+        hikari.templates.Template
+            An up-to-date view of this template.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the template is not found.
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        return await self.creator.app.rest.fetch_template(self.code)
+
+    async def edit(
+        self,
+        *,
+        name: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        description: undefined.UndefinedNoneOr[str] = undefined.UNDEFINED,
+    ) -> Template:
+        """Modify a guild template.
+
+        Parameters
+        ----------
+        name : hikari.undefined.UndefinedOr[str]
+            The name to set for this template.
+        description : hikari.undefined.UndefinedNoneOr[str]
+            The description to set for the template.
+
+        Returns
+        -------
+        hikari.templates.Template
+            The object of the edited template.
+
+        Raises
+        ------
+        hikari.errors.ForbiddenError
+            If you are not part of the guild.
+        hikari.errors.NotFoundError
+            If the guild is not found or you are missing the MANAGE_GUILD permission.
```

Comment:
```suggestion
            If the guild is not found or you are missing the `MANAGE_GUILD` permission.
```
Just so it renders a bit nicer in the docs. Not going to comment on all of them, but it goes for all the ones in this pr :)

Reply:
Done.